pkg/metrics/get_rate_limits_from_github.go

```diff
@@ -0,0 +1,63 @@
+package metrics
+
+import (
+	"context"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	remainingLimitsGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "github_remaining_limit",
+			Help: "number of requests remaining",
+		},
+		[]string{"type"},
+	)
+)
+
+var rateLimitsQuerySecondsBetween = 5
```

Comment:
The lowest scrape cycle for Prometheus is 1 minute so 5s granularity could be excessive. Perhaps we could align with the GITHUB_REFRESH setting

```suggestion
var rateLimitsQuerySecondsBetween = 60
```